src/lib.rs

```diff
@@ -1,8 +1,192 @@
 pub mod exchange;
 
+/*
+TODO:
+write unit tests for the code in this file.
+something else is code coverage not related to the file.
+get the PR done asap. Next week is the next project.
+
+ */
+use thiserror::Error;
+
+use crate::exchange::exchange::{Exchange, ExchangeError};
+use crate::exchange::price_feed::PriceFeed;
 use chrono::Utc;
+use rust_decimal::Decimal;
 use rust_decimal_macros::dec;
+use std::collections::HashMap;
+use thiserror::__private::AsDynError;
+use uuid::Uuid;
+
+#[derive(Debug, Clone, Error)]
+pub enum ExchangesError {
+    #[error("No exchange with the given exchange_id")]
+    InvalidExchangeId,
+}
+
+struct Exchanges {
+    exchanges: HashMap<String, Exchange>,
+}
+impl Exchanges {
+    pub fn new() -> Self {
+        Self {
+            exchanges: HashMap::new(),
+        }
+    }
+    pub fn create_new_exchange(&mut self) -> String {
+        let instance_id = Uuid::new_v4().hyphenated().to_string();
+        self.exchanges.insert(instance_id.clone(), Exchange::new());
+        instance_id
+    }
+    pub fn mut_unwrap_exchange_from_instance(
+        &mut self,
+        instance_id: &str,
+    ) -> Result<&mut Exchange, ExchangesError> {
+        if let Some(exchange) = self.exchanges.get_mut(instance_id) {
+            Ok(exchange)
+        } else {
+            Err(ExchangesError::InvalidExchangeId)
+        }
+    }
+    pub fn unwrap_exchange_from_instance(
+        &self,
+        instance_id: &str,
+    ) -> Result<&Exchange, ExchangesError> {
+        if let Some(exchange) = self.exchanges.get(instance_id) {
+            Ok(exchange)
+        } else {
+            Err(ExchangesError::InvalidExchangeId)
+        }
+    }
+    pub fn add_capital(
+        &mut self,
+        instance_id: &str,
+        symbol: &str,
+        amount: Decimal,
+    ) -> Result<(), ExchangesError> {
+        let exchange = self.mut_unwrap_exchange_from_instance(instance_id)?;
+        exchange.with_capital(vec![(symbol.to_string(), amount)]);
+        Ok(())
+    }
+    pub fn add_price_feed(
+        &mut self,
+        instance_id: &str,
+        symbol: &str,
+        interval: &str,
+        limit: i32,
+    ) -> Result<(), Box<dyn std::error::Error + Send + Sync + 'static>> {
```

Comment:
is the single `'` apostrophe at `static` - is that normal?

Reply:
Its rust's explicit lifetime annotations.
https://doc.rust-lang.org/rust-by-example/scope/lifetime/explicit.html

Reply:
gotcha - thanks!